src/System.Private.CoreLib/shared/System/Char.cs

```diff
@@ -931,19 +931,16 @@ public static string ConvertFromUtf32(int utf32)
             if (utf32 < UNICODE_PLANE01_START)
             {
                 // This is a BMP character.
-                return (char.ToString((char)utf32));
+                return ToString((char)utf32);
             }
 
-            unsafe
+            // This is a supplementary character.  Convert it to a surrogate pair in UTF-16.
+            utf32 -= UNICODE_PLANE01_START;
+            return string.Create(2, utf32, (span, value) =>
```

Comment:
This is makes it a little bit faster at the cost of quite bigger code (an extra method, delegate to hold the cached delegate, ...).

Given that this is low-performance API anyway - we have richer equivalent fast APIs on Rune, I think a better trade-off would to optimize to smallest codesize and make this just:

```
if (UnicodeUtility.IsValidUnicodeScalar())
    throw new ArgumentOutOfRangeException(nameof(utf32), SR.ArgumentOutOfRange_InvalidUTF32);
return Rune.UnsafeCreate(utf32).ToString();
```

Or something along these lines.

Reply:
Ok

Reply:
This does make it measurably slower than either of the two previous versions:
```
           Method |     Mean |     Error |    StdDev |  Gen 0 | Allocated |
----------------- |---------:|----------:|----------:|-------:|----------:|
 ConvertFromUtf32 | 21.73 ns | 0.5000 ns | 0.6323 ns | 0.0076 |      32 B |
```
An opportunity to improve Rune I guess.
cc: @GrabYourPitchforks